modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/CopyFormInstanceMVCActionCommand.java

```diff
@@ -126,6 +151,25 @@ protected ResourceBundle getResourceBundle(Locale locale) {
 	@Reference
 	protected SaveFormInstanceMVCCommandHelper saveFormInstanceMVCCommandHelper;
 
+	private void _removeExpirationDateDDMFormFieldValue(
+		DDMFormValues settingsDDMFormValues) {
+
+		for (DDMFormFieldValue ddmFormFieldValue :
+				settingsDDMFormValues.getDDMFormFieldValues()) {
```

Comment:
I believe this FOR is the same in the method `_removeExpirationDateDDMFormFieldValue()`;
If you analyze, the same variable `settingsDDMFormValuesCopy` is passed for both methods. Then, the FOR is the same.
I think we can unite both methods. What do you think?

Reply:
Hi @marcelabc ,
I refactored it to use only one FOR. I also noticed that we should use the time zone to check if the form is expired or not. So I did changes to accomplish that as well